cpp/src/arrow/builder.h

```diff
@@ -91,30 +91,31 @@ class ARROW_EXPORT ArrayBuilder {
   /// Set the next length bits to not null (i.e. valid).
   Status SetNotNull(int64_t length);
 
-  /// Allocates initial capacity requirements for the builder.  In most
-  /// cases subclasses should override and call their parent class's
-  /// method as well.
-  virtual Status Init(int64_t capacity);
+  /// \brief Ensure that at least the indicated number of elements can be appended to
+  /// the builder without a reallocation. Does not account for reallocations
```

Comment:
I find this wording a bit ambiguous. Does it mean that I can append `capacity` values to an already non-empty builder? I don't think so.

Reply:
Indeed ambiguous. I'm fixing